### PR TITLE
Feature movement horse

### PR DESCRIPTION
**Pull request by samuelloza**
I added the method for displayed possible movements of the horse 42c8ae2d3e8e650a8c41dda7f66a105e6e7a636d
I update chessBoard whit the new logic 2429bfb

### CODE REVIEW

File: src/main/java/chess/GetPossibleMovePiece.java [hunk 1]
```diff
@@ -27,6 +29,31 @@ void possibleMovement(final ArrayList<Position> arrayPositions, final int column
         }
     }
 
+    /**
+     * Add all movements of the Horse
+     *
+     * @param arrayPositions
+     */
+    public void possibleMovementHorse(final ArrayList<Position> arrayPositions, final int column, final int row) {
```
Comment (MirkoFer-622) [inline src/main/java/chess/GetPossibleMovePiece.java:37]: I think, you don't need composition between GetPossibleMovePiece adn Horse, only you could Implement this method into Horse class.

File: src/main/java/chess/GetPossibleMovePiece.java [hunk 2]
```diff
@@ -13,7 +15,7 @@
     void possibleMovement(final ArrayList<Position> arrayPositions, final int column, final int row, final int rowIncrement, final int columnIncrement) {
         int column1 = column;
         int row1 = row;
-        while ((row1 >= 0) && (column1 >= 0) && (row1 < LIMIT_OF_ARRAY) && (column1 < LIMIT_OF_ARRAY)) {
+        while ((row1 >= 0) && (column1 >= 0) && (row1 < BOARD_SIZE) && (column1 < BOARD_SIZE)) {
             if (isPossibleToContinue(piece.getChessBoard().getBoard()[column1][row1]) == -1) {
```
Comment (roht0515) [inline src/main/java/chess/GetPossibleMovePiece.java:19]: try to use constans, please

File: src/main/java/chess/GetPossibleMovePiece.java [hunk 3]
```diff
@@ -27,6 +29,31 @@ void possibleMovement(final ArrayList<Position> arrayPositions, final int column
...
+    /**
+     * Add all movements of the Horse
+     *
+     * @param arrayPositions
+     */
+    public void possibleMovementHorse(final ArrayList<Position> arrayPositions, final int column, final int row) {
+        int newColumn = column;
+        int newRow = row;
+        //int[] arrayMovementColumn = {2, 2, 1, -1, -2, -2, -1, 1};
+        //int[] arrayMovementRow = {1, -1, 2, 2, 1, -1, -2, -2};
```
Comment (roht0515) [inline src/main/java/chess/GetPossibleMovePiece.java:41]: delete this code if it is not needed

File: src/main/java/chess/GetPossibleMovePiece.java [hunk 4]
```diff
@@ -27,6 +29,31 @@ void possibleMovement(final ArrayList<Position> arrayPositions, final int column
...
+        int[] arrayMovementColumn = {NUMBER_TWO, NUMBER_TWO, NUMBER_ONE, -NUMBER_ONE, -NUMBER_TWO, -NUMBER_TWO, -NUMBER_ONE, NUMBER_ONE};
+        int[] arrayMovementRow = {NUMBER_ONE, -NUMBER_ONE, NUMBER_TWO, NUMBER_TWO, NUMBER_ONE, -NUMBER_ONE, -NUMBER_TWO, -NUMBER_TWO};
+        for (int i = 0; i < arrayMovementColumn.length; i++) {
+            newColumn = column + arrayMovementColumn[i];
+            newRow = row + arrayMovementRow[i];
+            if (newColumn >= 0 && newColumn < BOARD_SIZE && newRow >= 0 && newColumn < BOARD_SIZE) {
+                if (isPossibleToContinue(piece.getChessBoard().getBoard()[newColumn][newRow]) == -1) {
+                    arrayPositions.add(piece.getChessBoard().getBoard()[newColumn][newRow].getPosition());
+                } else if (isPossibleToContinue(piece.getChessBoard().getBoard()[newColumn][newRow]) == 0) {
```
Comment (roht0515) [inline src/main/java/chess/GetPossibleMovePiece.java:50]: please, try to use constants

File: src/main/java/chess/Horse.java [hunk 5]
```diff
@@ -2,15 +2,21 @@
 import java.util.ArrayList;
 
 class Horse extends Piece {
-    Horse(final Color color, final Position source) {
+
+    private GetPossibleMovePiece getPossibleMovePiece;
```
Comment (MirkoFer-622) [inline src/main/java/chess/Horse.java:6]: I comment it before, I think this composition is innecesary.

File: src/test/java/chess/HorseTest.java [hunk 6]
```diff
@@ -1,31 +1,117 @@
...
-import static org.junit.Assert.*;
-
-public class HorseTest {
-    @Test
-    public void testCreateHorse() {
-        Position pos = new Position(0, 0);
-        Horse piece = new Horse(Color.BLACK, pos);
-        assertNotNull(piece);
-    }
+    package chess;
```
Comment (MirkoFer-622) [inline src/test/java/chess/HorseTest.java:1]: There are trailing spaces before all class, true?

Comment (samuelloza) [reply src/test/java/chess/HorseTest.java:1]: Thks

File: src/main/java/chess/GetPossibleMovePiece.java (hunk 1 above)
Comment (samuelloza) [reply src/main/java/chess/GetPossibleMovePiece.java:37]: For moviments of the horse is necessary **"isPossibleToContinue"**, 
but this method is present in **GetPossibleMovePiece**.

I think if I implement in the **horse class** this is redundant.

File: src/main/java/chess/GetPossibleMovePiece.java (hunk 2 above)
Comment (samuelloza) [reply src/main/java/chess/GetPossibleMovePiece.java:19]: Thks

**Pull request merged**